src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -136,6 +137,9 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
                 // Disallow the import of targets/props from packages
                 argumentBuilder.Append(" /p:ExcludeRestorePackageImports=true ");
 
+                // Add all depenencies as top level restore projects if recursive is set
+                argumentBuilder.Append($" /p:RestoreNoRecurse={(!recursive).ToString()} ");
```

Comment:
instead of always appending a parameter, can't we do:
if(!recursive)
{
 //append parameter
}

or whatever the non-default behavior is.